### DISCUSSION TITLE
Contradictory `maxDuration` default value documentation

### DISCUSSION THREAD

**Discussion by pedro757** [vercel/next.js#68036] [Help] [answered]
### Summary

[This part of the docs](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#maxduration) state that there is no limitation when it comes to duration of sever-side logic

![image](https://github.com/user-attachments/assets/4ffad912-80ae-4f46-b3fa-cce4639db04e)

[This other part of the docs](https://vercel.com/docs/functions/runtimes#max-duration) say there's a 15s limit for the pro plan

![image](https://github.com/user-attachments/assets/ba7f057f-adb1-4a83-8472-609b4fbe6145)

So, which is it?

**Comment by icyJoseph** [comment, marked as answer]
Those are two different documentations though.

The first one, for Next.js, which doesn't have to hosted on Vercel, at all... And it is correct to say, that the framework itself doesn't impose such a duration.

The second one, is for Vercel's deployment platform, and theirs uses serverless functions, tasks, etc, and those are controlled by a TIMEOUT watcher, which kills the task after a given time. This applies to any work done in serverless functions in Vercel, be it API Routes, Route Handlers, revalidation, other frameworks, etc...

So, the first part is referring to a page level configuration, that a system such as part two, could use. IIRC, the build output includes a manifest JSON file, where the maxDuration is included, and a build/run system, can pick up on that to kill the rendering if it goes over a set time.

At least, that's how I am reading this.

**Reply by pedro757** [reply]
Thank you!
